api/v1beta1/azuremachine_webhook_test.go

```diff
@@ -698,6 +698,21 @@ func TestAzureMachine_ValidateUpdate(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "validtest: azuremachine.spec.networkInterfaces[0].subnetName is overwritten when empty",
```

Comment:
Thanks for the contribution! Just to clarify, is this supposed to (1) unset the `azureMachine.spec.subnetName` and (2) use that value to initialize `azuremachine.spec.networkInterfaces[0].subnetName`? If we want to do both, maybe we could clarify on the name of this test.